### PR TITLE
dasLLAMA: every serving shape steps batched - the CPU stack's deltanet and q4_0 forms, the block codecs' partial rope, the mirror cache grown to the streams, the rows-form drafts, and the image table's sb-only walks

### PR DESCRIPTION
**Pull request by borisbat**
**Behavior change: every dasLLAMA serving shape now steps batched - a deltanet hybrid, a q4_0 model and a block-codec KV session with partial rope no longer fall to a per-row step; a scheduler over more hybrid streams than four no longer dies; a `.dlim` of a model with Q5_1 or mx4 planes mints again.**

**Why.** A board cell served one row at a time is not a batched rate. After the last arc the CPU stack still stepped deltanet hybrids and q4_0 models per row, the Metal driver declined a partial-rope model on q8_0 / tq4 KV, a batch over five or more hybrid sessions evicted its own recurrent state (and then faulted on the device), the tq4 codec decoded garbage on every 256-wide head, the gemma-4-26B image mint failed its map-back, and the self-speculative round drafted one stream per dispatch (spec @4 at 0.74 of plain on the 0.8B).

**What changes.**
- An arch names its batched layer in `ArchBlocks.attn_batch`; qwen35 serves its recurrent layers through `deltanet_batch` (rows GEMMs, then the delta rule per row and head against the session's own state) and its gated attention through the standard batched attention with the gate folded into the shared prefix and suffix.
- `matmul_q4_batch` gives the q4_0 tier a batched GEMM, and the four per-row gates on q4_0 (batch, prefill, embd prefill, MTP) are gone.
- The q8_0 and tq4 rope-store kernels take a rotated prefix (`rot`), single and batched, so the block codecs serve partial-rope carriers; the tq4 signs table covers heads to 512.
- The deltanet mirror cache grows to the scheduler's stream count through the `gpu_dn_room` seam, and the rows planes pad to the four-row tile.
- The speculative round drafts every warm stream's draft i as one rows step through the draft head (`mtp_draft_rows`); the rows prelude, the classifier tail and the row lander are shared with the verify.
- The image writer and reader skip the per-format table's non-`kq_sb` slots, whose planes live top-level under the same section name.
- `harness/batch_decline_census.das` (the batched-step decline census) and `harness/forced_feed_probe.das` (per-step CPU-vs-GPU logits under a forced feed) are the two instruments, both on `daslib/clargs`.

**Observable behavior.**
- Qwen3.5-0.8B Q8 on the CPU stack at four streams: per-row steps -> the CPU batched stack, tg128@4 517 against llama.cpp's 471 at `-ngl 0`.
- Qwen3.5-0.8B Q8 on q8_0 or tq4 KV through the Metal driver: `kv_dtype` decline -> served batched; a tq4 session on a 256-wide head: logits maxd 6-13 with argmax flips -> 0.2-0.6, tokens exact.
- Nine hybrid streams: `dn_state` decline then a device fault -> served (the 0.8B at twelve too).
- Qwen3.5-0.8B-MTP spec @4 (released exe, idle box): 0.74 of the plain row -> 0.91, and above llama.cpp's plain row (890 vs 811); the 27B-MTP reads 0.91 of plain within a 12% cv.
- gemma-4-26B-A4B Q4_K_M `.dlim` mint: map-back fails on two `q51q` sections -> mints and serves.
- The Metal catalog rows the E-series, shared-expert and deltanet arms unlocked, all served batched (tg128@4 ours / llama.cpp): Qwen3.5-4B 295 / 248, Qwen3-30B-A3B 256 / 215, Qwen3.6-35B-A3B 232 / 198.
- `test_vulkan_moe_cm2.das` on a build with no Vulkan module: compile failure -> skips.

**Where to look.** `dasllama_arch_qwen35.das` (`deltanet_batch`, `attention_qwen35_batch`, `dn_forward_only_guard`), the `rot` arms of `dasllama_metal_kernels.das`'s four rope-store classes, `dn_mirror_room` and `mr` in `dasllama_metal_common.das` / `dasllama_metal_decode.das`, and the rows-form drafts (`mtp_draft_rows`, `encode_draft_rows_step`, `enc_draft_head_rows`, `enc_cls_tail_rows`, `land_draft_row`) beside `encode_verify_step`.

<details>
<summary>Validation, claims, ledger</summary>

### Validation
- Preflight full tier ran once: the fast tier's one red was a non-ASCII glyph in an architecture doc (fixed, targeted gate green); the docs lane wanted the new facade def grouped on the reference page (done); the C++, interpreter, JIT and AOT lanes each ran green through `--only`.
- The rule documents this branch edits each closed two dragon passes; the module and GPU checklists came off the 300-line split cap by moving three rules to the checklists their diffs already route to.
- Generated Metal source per stamp against origin/master (the `*_msl` globals dumped from both trees): `metal_rope_store_q8_msl`, `metal_rope_store_b_q8_msl`, `metal_rope_store_tq4_msl` and `metal_rope_store_b_tq4_msl` differ by the `rot` arm (the rotated prefix and its identity pass-through), pinned by the rope gate cells at head 256 rot 64; `metal_fwht_unsign_msl` and `metal_rope_store_h16_msl` are empty diffs.
- The tq4 signs table's 512-float ceiling read from the emitted artifact: `metal_rope_store_tq4_msl_tgmem` and `metal_rope_store_b_tq4_msl_tgmem` both read `0x800` (2048 bytes = 512 floats), the stamp entry taking `threadgroup char * shmem [[threadgroup(0)]]` cast to `threadgroup float * buf`.
- `IMAGE_VERSION` stays at 39 with the layout stamp re-hashed: no `.dlim` a previous build could load changes content or meaning - the only writer change is in images that declined on duplicate section names, and the dropped map-back alias of a non-sb table slot is read by no serving path (`kq_planes_of` panics for a non-sb format).
- The tower gate the metal_common change owes (`--suite image --arm mtower`): 44 passed, 0 failed - the whisper tiny Metal tower blocks, the GPU cross-KV decoder and the parakeet absence cell ran; the cells whose carriers are not stocked here skipped loudly.
- The kernel cells at the whole head named as the prefix (`rot == head_size`, what every full-rope layer sends) byte-match the default on all four block-codec rope-store kernels (`--suite kernels`, 7 files green).
- Local Metal gates on the M5 Max: `test_batch_decode` 25/25, `test_metal_batched_row` 13/14 (the shared-expert MoE cell is large-tier), the MTP parity arms (`mtp-vff`, `mtp-dff`, the 0.8B's ff/ffk/count/count8/sampled), the support matrix's `fam-qwen35` row over f16/q8_0/tq4, the rope kernel cells at head 256 rot 64, and the module's `REVIEW.das` gate; the runner's `--changed` set (144 files) green.
- The external reviewer round read the changed set and reported no findings; its own test execution did not run (its sandbox is read-only, so the runner could not create its cache).
- Vulkan items (followup_vulkan 71 and 76, media sessions in device mode) were not run - no Vulkan device on this box.
- The board is not re-minted here; the ledger's new readings are direction-grade (two processes, or the `-jit` script) except the released-exe MTP rows on the idle box.

### Claims - stated, not tested
- The MTP rows-form drafts on the K-quant 27B carrier under `DASLLAMA_PARITY_FULL=1`: the same code path as the 0.8B's arms (which are exact), verified by the ledger's rate reading only; a break would show as a spec row below plain or a joint-invariance mismatch on that carrier.
- Parity of the changed paths on carriers outside both boards (an IQ-quant file on tq4 / q8_0 KV): the forced-feed probe is the instrument; run here on the hybrid Q8 and gemma-2-2b Q4_K_M only.
- The rows-form drafts' pad rows (rows past the streams, up to the four-row tile) carry row 0's route and inputs; the joint-invariance cell stays green with the pad entries zeroed or left stale (negative-controlled), so the clone is hygiene against reading unwritten memory, not a pinned behaviour.
- The step-side `dn_mirror_room(nrows)` in the batched driver is redundant on every scheduler-driven path (the scheduler seam sizes the cache first); it covers a direct `eval_batch` caller and no cell distinguishes it.
- `attention_gpu_prefill` now passes the model's `q_gated` flag (was hard-coded off); the only instrument is `test_gpu_resident_hybrid` on a box whose resident (Vulkan) tier arms - not run here. A break would show as that file's forced-feed bar failing on Qwen3.5-0.8B.
- `deltanet_batch`'s f32-on-disk beta/alpha arm (`dn_ba_f32`) is reached only by the 22 GB qwen35moe carrier (`DASLLAMA_PARITY_FULL=1`, `--suite matrix --family qwen35moe`) - not run here.
- Error arms with no failure-injection fixture: `kn_buffer`'s null-slot panic, the rows drafts' mirror-guard decline (a stale watermark / cap / codec), the rows step's dispatch-failure decline, and the round's per-stream plain fallback when a draft declines.

### Not done
- The board (`performance/records/{box}.json`) is not re-minted and no row is withdrawn: the affected `tg128@4` / `tg128` / `pp` rows describe the pre-arc build. A mint mid-arc re-baselines every later reading against a moving tree; the board re-mints once, after the arc lands, on the idle box. The ledger's readings above are direction-grade except the released-exe MTP rows.
- followup_metal 22 (the deltanet state arena - one dispatch a stage) - perf only, the hybrids already serve batched.
- followup_vulkan 71 (the N-row command's declined layer kinds), 76 (rows sampled one after another) and media sessions in device mode - need a Vulkan box.
- The folds the dupe audit named TEMPLATABLE (the q4 batched nest beside the other batched GEMMs; the deltanet batch/prefill prologue-epilogue pair) wait for the post-arc dedup pass.

</details>

### CODE REVIEW

File: modules/dasLLAMA/dasllama/dasllama_common.das [hunk 1]
```diff
@@ -3735,6 +3739,8 @@ def mm_b(var y : array<float>; t : Model; woff : int64; x : array<float>; var xq
         } else {
             mm_b_q8_pre(y, t, woff, xqb, xsb, n, d, npos)
         }
+    } elif (t.quant == QuantMode.q4_0) {
+        matmul_q4_batch(y, t.q4blob, t.q4scales, woff, x, n, d, npos)
```
[the remark this answers is not in the document]
Comment (borisbat) [reply modules/dasLLAMA/dasllama/dasllama_common.das:3747]: Accepted - the leading comment now reads that q4_0 rides the weight-stationary nest over the GEMV's dot (820733228).
